File: 0073-set-matrix-zeroes/0073-set-matrix-zeroes.java
class Solution {
    public void setZeroes(int[][] matrix) {
        int row=matrix.length;
        int col=matrix[0].length;
        int ans1[]=new int[row];
        int ans2[]=new int[col];
        Arrays.fill(ans1,-1);
        Arrays.fill(ans2,-1);
        for(int i=0;i<row;i++){
            for(int j=0;j<col;j++){
                if(matrix[i][j]==0){
                    ans1[i]=0;
                    ans2[j]=0;
                }
            }
        }
        for(int i=0;i<row;i++){
            for(int j=0;j<col;j++){
                if(ans1[i]==0 || ans2[j]==0){
                    matrix[i][j]=0;
                }
            }
        }
    }
}

File: 0073-set-matrix-zeroes/0073-set-matrix-zeroes.js

var setZeroes = function(matrix) {
    let rows = matrix.length;
    let cols = matrix[0].length;
    let rowZero = new Array(rows).fill(false);
    let colZero = new Array(cols).fill(false);
    for (let i = 0; i < rows; i++) {
        for (let j = 0; j < cols; j++) {
            if (matrix[i][j] === 0) {
                rowZero[i] = true;
                colZero[j] = true;
            }
        }
    }
    for (let i = 0; i < rows; i++) {
        for (let j = 0; j < cols; j++) {
            if (rowZero[i] || colZero[j]) {
                matrix[i][j] = 0;
            }
        }
    }
};